### PR TITLE
sched: Fix CPU affinity issues in SMP

### PR DESCRIPTION
**Pull request by vito-chl**
In "nxsched_nexttcb": the task may not running on this_cpu, and rtrtcb->affinity(the affinity of the task in g_readytorun) may not include the current cpu which should be the tcb->cpu.

In "nxsched_remove_readytorun": as the problem in "nxsched_nexttcb", the logical may choose the head of g_readytorun(the greatest priority) task as the next while ignoring the cpu affinity of it.

### CODE REVIEW

File: sched/sched/sched_removereadytorun.c [hunk 1]
```diff
@@ -209,20 +209,16 @@ bool nxsched_remove_readytorun(FAR struct tcb_s *rtcb)
 
       if (rtrtcb != NULL && rtrtcb->sched_priority >= nxttcb->sched_priority)
         {
-          FAR struct tcb_s *tmptcb;
-
           /* The TCB at the head of the ready to run list has the higher
            * priority.  Remove that task from the head of the g_readytorun
            * list and add to the head of the g_assignedtasks[cpu] list.
```
Comment (masayuki2009) [inline sched/sched/sched_removereadytorun.c:214]: I think we need to modify these comments as well.
Please change the comments and merge them into this commit.

File: sched/sched/sched_setpriority.c [hunk 2]
```diff
@@ -69,12 +68,12 @@ static FAR struct tcb_s *nxsched_nexttcb(FAR struct tcb_s *tcb)
    * then use the 'nxttcb' which will probably be the IDLE thread.
    */
 
-  if (!nxsched_islocked_global() && !irq_cpu_locked(cpu))
+  if (!nxsched_islocked_global() && !irq_cpu_locked(this_cpu()))
     {
       /* Search for the highest priority task that can run on this CPU. */
```
Comment (masayuki2009) [inline sched/sched/sched_setpriority.c:73]: I think this comment should be modified as well.

File: sched/sched/sched_setpriority.c [hunk 3]
```diff
@@ -69,12 +68,12 @@ static FAR struct tcb_s *nxsched_nexttcb(FAR struct tcb_s *tcb)
    * then use the 'nxttcb' which will probably be the IDLE thread.
    */
```
Comment (masayuki2009) [inline sched/sched/sched_setpriority.c:70]: I think the above comments `if the affinity mask includes the current CPU` should be modified as well.

**Review by masayuki2009 — CHANGES_REQUESTED** [review]

**Comment by vito-chl** [comment]
@masayuki2009 Done!

**Review by masayuki2009 — APPROVED** [review]
LGTM

**Comment by jerpelea** [comment]
LGTM

**Pull request merged**